datasets/timit_asr/timit_asr.py

```diff
@@ -0,0 +1,171 @@
+
+
+class TimitASRConfig(datasets.GeneratorBasedBuilder):
+    """TimitASR dataset."""
+
+    BUILDER_CONFIGS = [
+        TimitASRConfig(name="clean", description="'Clean' speech."),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "file": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "phonetic_detail": datasets.Sequence(
+                        {
+                            "start": datasets.Value("int64"),
+                            "stop": datasets.Value("int64"),
+                            "utterance": datasets.Value("string"),
+                        }
+                    ),
+                    "word_detail": datasets.Sequence(
+                        {
+                            "start": datasets.Value("int64"),
+                            "stop": datasets.Value("int64"),
+                            "utterance": datasets.Value("string"),
+                        }
+                    ),
+                    "dialect_region": datasets.Value("string"),
+                    "sentence_type": datasets.Value("string"),
+                    "speaker_id": datasets.Value("string"),
+                    "id": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=("file", "text"),
+            homepage=_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download_and_extract(_URL)
+
+        train_csv_path = os.path.join(archive_path, "train_data.csv")
+        test_csv_path = os.path.join(archive_path, "test_data.csv")
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"data_info_csv": train_csv_path}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"data_info_csv": test_csv_path}),
+        ]
+
+    def _generate_examples(self, data_info_csv):
+        """Generate examples from TIMIT archive_path based on the test/train csv information."""
+        # Extract the archive path
+        data_path = os.path.join(os.path.dirname(data_info_csv).strip(), "data")
+
+        # Read the data info to extract rows mentioning about non-converted audio only
+        data_info = pd.read_csv(data_info_csv)
+        data_info = data_info.loc[(data_info["is_audio"] == True) & (data_info["is_converted_audio"] == False)]
+
+        # Iterating the contents of the data to extract the relevant information
+        for audio_idx in range(data_info.shape[0]):
+            audio_data = data_info.iloc[0]
+
+            # extract the path to audio
+            wav_path = os.path.join(data_path, *(audio_data["path_from_data_dir"].split("/")))
+
+            # extract transcript
+            with open(wav_path.replace(".WAV", ".TXT"), "r") as op:
```

Comment:
Each time you use `open`, please specify the encoding. For example `encoding="utf-8"`.
This is useful especially for windows users, and this should fix the CI error.

Also don't forget to do `make style` do apply code formatting :)